File: Leetcode/1200. Minimum Absolute Difference.py
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        res=[]
        arr.sort()
        mini=10000000
        for i in range(len(arr)-1):
            if(arr[i+1]-arr[i]==mini):
                res.append([arr[i],arr[i+1]])
            elif arr[i+1]-arr[i]<mini:
                mini=arr[i+1]-arr[i]
                res=[[arr[i],arr[i+1]]]
        return res
        

File: Leetcode/1200. Minimum Absolute Difference.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        sort(arr.begin(),arr.end());
        int mini=INT_MAX;
        int n=arr.size();
        unordered_map<int,vector<vector<int>>> m;
        for(int i=0;i<n-1;i++){
            mini=min(mini,arr[i+1]-arr[i]);
            vector<int> t(2);
            t={arr[i],arr[i+1]};
            m[arr[i+1]-arr[i]].push_back(t);
        }
        return m[mini];
    }
};
